src/core/net/srp_client.cpp

```diff
@@ -1160,6 +1181,27 @@ void Client::ProcessResponse(Message &aMessage)
         GrowRetryWaitInterval();
         SetState(kStateToRetry);
         InvokeCallback(error);
+
+#if OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE && OPENTHREAD_CONFIG_SRP_CLIENT_SWITCH_SERVER_ON_FAILURE
+        if ((error == kErrorDuplicated) || (error == kErrorSecurity))
```

Comment:
`kErrorDuplicated` this usually indicates we should use a new name. Why switching to a new server will resolve the error?

Reply:
The `kErrorDuplicated` can be  helpful in situation where another server is still remembering the service (which client registered with it earlier) and not letting it be registered with a new server (particularly when there is no replication between servers). In such a case,  on the client we try to switch the servers till we get back to same server that is keeping/remembering the name.

Reply:
What if there is truly a duplicate? [draft-ietf-dnssd-srp-09 Section 2.2.5.2](https://datatracker.ietf.org/doc/html/draft-ietf-dnssd-srp-09#section-2.2.5.2) suggests changing the existing name to resolve the conflict. When would we/the client decide to apply this logic?

Reply:
We have a callback from `Client` to reprot the failures to the caller and the `SelectNextServer()` is called after invoking the callback.

In the case of an actual duplicate, I think we can expect the caller to take action and change the name (mainly since the names are really selected by the caller).

Reply:
But how would the client know whether to wait to retry a new server or change the name?

Reply:
Some instance names are unique by nature and can't be changed, e.g. CHIP ID. 
Other instances names can be changed to a different name, e.g. `Printer` -> `Printer (2)`. 
Seems it's better to have different strategy for different kind of instances?

Reply:
> But how would the client know whether to wait to retry a new server or change the name?

I think the user of `Srp::Client` can be fully aware of what's happening (from different callback and API calls) and apply  different strategies based on its use-case (as @simonlingoogle  mentioned different names may need different strategies):
- Caller can register a callback to get outcome when registration fails (e.g. server rejects).
- Caller should know whether it has enabled auto-start mode or not (and whether switch server feature is enabled or not)
- It can also track the server changes through the auto-start callback (which notifies caller whenever auto-start selects a new server).

As a simple strategy example, caller can implement a rule of trying to change the name after some `x` back-to-back server rejections (as duplicate).
Or a more complex case, caller can track the server changes and try to change the name when it sees that we get back to the first server again (exhaust all server options and get rejected by all).

I think later on, if we see there is a common useful strategy, we can consider adding it in `Srp::Clinet` directly (i.e. have a separate callback to notify caller that `Srp::Clinet` recommends a name change).